### PR TITLE
Add info about filters to documentation for /conversations endpoint. …

### PR DESCRIPTION
**Pull request by yuriykurylenko**
…AB-8610

API Documentation for https://autobutler.atlassian.net/browse/AB-8610

### CODE REVIEW

**Comment by dipth** [comment]
Personally I would argue that these filters should instead be different restful collections instead:

**Comment by yuriykurylenko** [comment]
My main concern here is in the case when we will have a need to filter on some other field in the future, as well.

Let's imagine we have 2 fields to filter on state and subject_type.
We can have 4 states: all, unread, unanswered and answered,
and 2 subject types: Job and Offer.

If we are to follow the pattern, described above, we will need to have 4*2 = 8 endpoints to cover all the possible combinations of filters, and this number will only grow with growing number of filter or  options in each single filter.

While when using query params, we can still have only 1 endpoint, smth. like this:
/conversations?state=unread&subject_type=Job.

So the approach with using query params is just  more scalable..

**Comment by dipth** [comment]
I would then argue that if you have different subject types, the endpoints should be as follows:

```
/api/v2/mechanics/jobs/job_id/conversations
/api/v2/mechanics/jobs/job_id/conversations/unread
/api/v2/mechanics/jobs/job_id/conversations/unanswered
/api/v2/mechanics/jobs/job_id/conversations/answered
/api/v2/mechanics/offers/offer_id/conversations
/api/v2/mechanics/offers/offer_id/conversations/unread
/api/v2/mechanics/offers/offer_id/conversations/unanswered
/api/v2/mechanics/offers/offer_id/conversations/answered
```
Since a list of conversations pertaining to a job is not interchangeable with a list of conversations pertaining to an offer. Always be thinking restful :)

**Comment by yuriykurylenko** [comment]
Yes, but what if you need to add the third filter - a car make, for instance? ;)

File: source/includes/_conversations.md [hunk 1]
```diff
@@ -5,13 +5,13 @@
 > To obtain a list of conversations, use the following code:
 
 ```shell
-curl -X "GET" "https://www.autobutler.dk/api/v2/mechanics/conversations?page=1" \
+curl -X "GET" "https://www.autobutler.dk/api/v2/mechanics/conversations?page=1&filter=all" \
```
Comment (andreorvalho) [inline source/includes/_conversations.md]: There is no filter all anymore :)

File: source/includes/_conversations.md [hunk 2]
```diff
@@ -5,13 +5,13 @@
...
      -H "Authorization: token"
 ```
 
 ```javascript
 jQuery.ajax({
-    url: "https://www.autobutler.dk/api/v2/mechanics/conversations?page=1",
+    url: "https://www.autobutler.dk/api/v2/mechanics/conversations?page=1&filter=all",
```
Comment (andreorvalho) [inline source/includes/_conversations.md]: same

File: source/includes/_conversations.md [hunk 3]
```diff
@@ -63,15 +63,24 @@ jQuery.ajax({
...
 
 ### HTTP Request
 
-`GET https://www.autobutler.dk/api/v2/mechanics/conversations?page={page}`
+`GET https://www.autobutler.dk/api/v2/mechanics/conversations?page={page}&filter={filter}`
+
+#### Request params
+* *page* - specifies a page number for queried conversations.
+* *filter* - specifies a filter for the queried conversations. Possible values:
+  * `all` - queries all conversations for the mechanic
```
Comment (andreorvalho) [inline source/includes/_conversations.md]: same

File: source/includes/_conversations.md (hunk 1 above)
Comment (yuriykurylenko) [inline source/includes/_conversations.md]: Why???

**Comment by dipth** [comment]
> Yes, but what if you need to add the third filter - a car make, for instance? ;)

Then you're getting into the realm of searching which should probably have it's own set of endpoints

**Pull request merged**